apps/google-analytics-4/lambda/src/lib/verify-signed-request-middleware.ts

```diff
@@ -0,0 +1,87 @@
+import * as NodeAppsToolkit from '@contentful/node-apps-toolkit';
+import {
+  CanonicalRequest,
+  ContentfulContextHeader,
+  ContentfulHeader,
+} from '@contentful/node-apps-toolkit/lib/requests/typings';
+import express from 'express';
+import { InvalidSignature } from '../lib/errors/invalid-signature';
+import { UnableToVerifyRequest } from '../lib/errors/unable-to-verify-request';
+import { IncomingHttpHeaders } from 'http';
+
+export const verifySignedRequestMiddleware = (
+  req: express.Request,
+  _res: express.Response,
+  next: express.NextFunction
+) => {
+  const signingSecret = (process.env.SIGNING_SECRET || '').trim();
```

Comment:
I think we should (in a follow up PR, not now) formalize our env var usage via a config instead of hard coding here. This is what I did for the frontend https://github.com/contentful/apps/blob/apps/ga4/apps/google-analytics-4/frontend/src/config.ts

Specifically we should fail at start if the env var is missing rather than attempting to fail gracefully as we're doing here. (This approach will just nicely give customers signing errors instead of telling us developers that we messed up our deploy 😉 )

Reply:
Agreed!

Reply:
Created a ticket: https://contentful.atlassian.net/browse/INTEG-128